src/components/menu/menu.component.js

```diff
@@ -1,132 +1,81 @@
-import React, { useRef, useCallback, useState } from "react";
+import React, { useEffect, useState, useCallback } from "react";
 import PropTypes from "prop-types";
-import {
-  StyledMenuWrapper,
-  StyledMenuItemsWrapper,
-  StyledMenuItem,
-} from "./menu.style";
+
+import { StyledMenuWrapper } from "./menu.style";
+import { menuKeyboardNavigation } from "./__internal__/keyboard-navigation";
 import Events from "../../utils/helpers/events";
-import MenuItem from "./menu-item";
 
-const Menu = ({ menuType = "light", children }) => {
-  const childrenArray = Array.isArray(children) ? children : [children];
-  const filteredChildren = childrenArray.filter(
-    (child) => child.type === MenuItem
-  );
-  const menuItemsRefs = useRef(
-    filteredChildren.map((child) => child.ref || React.createRef())
-  );
-  const actualFocusedItemIndex = useRef();
-  const [isOpenSubmenu, setIsOpenSubmenu] = useState(null);
-  const [isOpenByArrowLeftOrRight, setOpenByArrowLeftOrRight] = useState(false);
-  const setFocusToElement = useCallback((event, index) => {
-    if (event) {
-      event.preventDefault();
-    }
+const MenuContext = React.createContext({});
 
-    actualFocusedItemIndex.current = index;
-    const isFocusedOnLastMenuItem = index === menuItemsRefs.current.length;
-    const isFocusedOnFirstMenuItem = index === -1;
-    if (isFocusedOnLastMenuItem) {
-      actualFocusedItemIndex.current = 0;
-    } else if (isFocusedOnFirstMenuItem) {
-      actualFocusedItemIndex.current = menuItemsRefs.current.length - 1;
-    }
+const Menu = ({ menuType = "light", children, ...rest }) => {
+  const [focusedItemIndex, setFocusedItemIndex] = useState(undefined);
+  const [openSubmenuIndex, setOpenSubmenuIndex] = useState(undefined);
 
-    menuItemsRefs.current[actualFocusedItemIndex.current].current.focus();
-  }, []);
+  const reset = () => {
```

Comment:
This could be moved inside the scope of `onClickOutside` as it's not used anywhere else